### PR TITLE
Teacher registration warning

### PR DESCRIPTION
**Pull request by hannah-whelan**
https://github.com/isaaccomputerscience/isaac-cs-issues/issues/222

### CODE REVIEW

File: src/app/components/pages/TeacherRegistration.tsx [hunk 1]
```diff
@@ -21,10 +21,38 @@ import useRegistration from "../hooks/useRegistration";
 import { RegistrationSubmit } from "../elements/inputs/RegistrationSubmit";
 import ReCAPTCHA from "react-google-recaptcha";
 import { Recaptcha } from "../elements/inputs/Recaptcha";
+import { Link } from "react-router-dom";
 
 const metaDescriptionCS =
   "Sign up for a free account and get powerful GCSE and A Level Computer Science resources and questions. For classwork, homework, and revision.";
 
+export const AreYouATeacher = ({ confirmTeacher }: { confirmTeacher: (arg0: boolean) => void }) => {
```
Comment (jmgq) [inline src/app/components/pages/TeacherRegistration.tsx:29]: This is only being used locally, so it shouldn't be exported.

Comment (jmgq) [inline src/app/components/pages/TeacherRegistration.tsx:29]: I also think that the `confirmTeacher` name could be improved. Based on a [convention](https://stackoverflow.com/a/60048346) that I've seen in other projects, this could be called `onTeacherConfirmed`, for instance.

That way when you use this later on, it reads nicely:
```typescript
<AreYouATeacher onTeacherConfirmed={() => setRegistrationStep(1)} />
```

File: src/app/components/pages/TeacherRegistration.tsx [hunk 2]
```diff
@@ -21,11 +21,39 @@ import useRegistration from "../hooks/useRegistration";
...
+        className="mb-4"
+      />
+      <Col>
+        <p className="mt-5 mx-auto col-lg-10">
+          You will be required to provide evidence of being a teacher before your teacher account will be activated. If
+          you are not listed on your school&apos;s website we will be required to contact your school directly.
+        </p>
+        <div className="text-center mb-1 mt-4">
+          <Button onClick={() => onTeacherConfirmed()} className="m-2 m-lg-4 btn btn-success border-0">
+            Yes I am a teacher
```
Comment (jmgq) [inline src/app/components/pages/TeacherRegistration.tsx:45]: Small one, but for clarity it would be good to add a comma after the "Yes". So: "Yes, I am a teacher". The same for the student: "No, I am a student".

File: src/app/components/pages/TeacherRegistration.tsx [hunk 3]
```diff
@@ -108,7 +136,7 @@ export const TeacherRegistrationTerms = ({ acceptConditions }: { acceptCondition
 };
 
 // TODO: useLocation hook to retrieve email/password when upgrading react router to v6+
-export const TeacherRegistrationBody = () => {
+export const TeacherRegistrationForm = () => {
```
Comment (jmgq) [inline src/app/components/pages/TeacherRegistration.tsx:139]: Now that you've edited this line, you could also remove the `export`.

File: src/test/pages/TeacherRegistration.test.tsx [hunk 4]
```diff
@@ -28,16 +33,32 @@ describe("Teacher Registration", () => {
     });
   };
 
-  it("On page load the teacher conditions are visible and accept button needs to be pressed in order to view the form", async () => {
+  it("it displays teacher confirmation screen, then teacher conditions, then registration form", async () => {
```
Comment (jmgq) [inline src/test/pages/TeacherRegistration.test.tsx:36]: Another really small one. Don't need to change this if you don't want, but the reason Jest gives you `it` as an alias of `test` is so that you can write test names that are readable in plain English. So this one could be:
```typescript
it("should display..."
```

So it reads as "It should display". The way it's done now it reads as "It it displays". I tend to write tests that start with "should", as it forces me to think of a good name, but you don't have to follow that pattern. But bear in mind that when you use "it", people are going to read that as part of the whole sentence. As I say, you don't have to use "should", for instance what you've done in a test below also works, as it reads perfectly:
```typescript
it("redirects to student registration page if user is not a teacher", async () => {
```

File: src/test/pages/TeacherRegistration.test.tsx [hunk 5]
```diff
@@ -19,6 +19,11 @@ import { registrationMockUser, registrationUserData } from "../../mocks/data";
 const registerUserSpy = jest.spyOn(actions, "registerUser");
 const upgradeAccountSpy = jest.spyOn(actions, "upgradeAccount");
 
+const clickThroughOptions = async () => {
```
Comment (jmgq) [inline src/test/pages/TeacherRegistration.test.tsx:22]: This name should reflect the fact that you're following the Teacher path, just to make this more descriptive.

Comment (hannah-whelan) [reply src/test/pages/TeacherRegistration.test.tsx:22]: Renamed this one to `confirmTeacherAndAcceptTerms`

File: src/test/pages/TeacherRegistration.test.tsx (hunk 4 above)
Comment (hannah-whelan) [reply src/test/pages/TeacherRegistration.test.tsx]: Yeah that makes sense - edited several test names relating to registration and I'll keep an eye on this one as I edit or add to other tests.

**Review by jmgq — APPROVED** [review]
